tracer/src/Datadog.Trace/LifetimeManager.cs

```diff
@@ -79,6 +81,7 @@ private void CurrentDomain_ProcessExit(object sender, EventArgs e)
         private void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs e)
         {
             Log.Warning("Application threw an unhandled exception: {Exception}", e.ExceptionObject);
+            CurrentException = e.ExceptionObject as Exception;
```

Comment:
I know why you did this, because you didn't want to change the signature of RunShutdownTasks, but I think that's what we should do 😉 

I'm fine with tackling that myself in a subsequent PR though if you like?

Reply:
Ok, let's do that in a subsequent PR then 👍🏻 thanks!

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/5557